Experimental/DistributedAuthoritySample/Assets/Scripts/UI/IngameUI/PlayersTopUIController.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using System.Collections.Generic;
+using Unity.Netcode;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace Unity.Multiplayer.Samples.SocialHub.UI
+{
+    /// <summary>
+    /// UI controller which displays nameplate and mic icon on top of each connected player.
+    /// </summary>
+    public class PlayersTopUIController : MonoBehaviour
+    {
+        [SerializeField]
+        UIDocument m_UIDocument;
+
+        [SerializeField]
+        VisualTreeAsset m_NameplateAsset;
+
+        [SerializeField]
+        float m_PanelMinSize = 0.8f;
+
+        [SerializeField]
+        float m_PanelMaxSize = 1.1f;
+
+        [SerializeField]
+        float m_DisplayYOffset = 1.3f;
+
+        [SerializeField]
+        Camera m_Camera;
+
+        List<PlayerHeadDisplay> m_PlayerHeadDisplayPool = new();
+
+        List<Tuple<NetworkObject, PlayerHeadDisplay>> m_PlayersToDisplayMap = new();
+
+        VisualElement m_Root;
+
+        const int k_PoolSize = 12;
+
+        void OnEnable()
```

Comment:
What should be cleared on OnDisable()?

Reply:
I am clearing the pooled objects now and remove it from UiDocument hierarchy